sdk/python/lib/pulumi/runtime/rpc_manager.py

```diff
@@ -25,26 +26,24 @@ class RPCManager:
     outstanding RPCs.
     """
 
-    zero_cond: asyncio.Condition
+    count: int
     """
-    zero_cond is notified whenever the number of active RPCs transitions from
-    one to zero.
+    The number of active RPCs.
     """
 
-    count: int
+    unhandled_exception: Optional[Exception]
     """
-    The number of active RPCs.
+    The first unhandled exception encountered during an RPC, if any occurs.
     """
 
-    exception_future: asyncio.Future
+    exception_traceback: Optional[Any]
     """
-    Future that is resolved whenever an unhandled exception occurs.
+    The traceback associated with unhandled_exception, if any.
     """
 
     def __init__(self):
-        self.zero_cond = asyncio.Condition()
         self.count = 0
-        self.exception_future = asyncio.Future()
+        self.unhandled_exception = None
```

Comment:
Should `exception_traceback` also be initialized?

Reply:
If only for the sake of consistency, yes (which is why I initialized `unhandled_exception`).